FAQ.md

```diff
@@ -165,3 +165,48 @@ together in a number of configurations:
+many architectures is actively developed and maintained by a large community. In fact [PNaCl](http://gonacl.com)
+already uses LLVM as a basis for its binary format. However the goals and requirements that LLVM was designed
+to meet are subtly mismatched with those of WebAssembly.
+
+WebAssembly has several requirements and goals for its IR and binary encoding:
+ * Portability: The IR must be the same for every machine architecture.
+ * Stability: The IR and binary encoding must not change over time (or change only in ways that can
+   be kept backward-compatible).
+ * Small encoding: The representation of a program should be as small as possible for transmission over
+   the Internet.
+ * Fast decoding: The binary format should be fast to decompress and decode for fast startup of programs.
+ * Fast compiling: The IR should be fast to compile (and suitable for either AOT- or JIT-compilation) for fast
+   startup of programs.
+ * Minimal [nondeterminsim](Nondeterminism.md): The behavior of programs should be as predictable and
+   deterministic as possible (and should be the same on every architecture, a stronger form of the
+   portability requirement stated above).
+
+LLVM IR is meant to make compiler optimizations easy to implement, and to represent the constructs
+and semantics required by C, C++, and other languages on a large variety of operating systems and
+architectures. This means that by default the IR is not portable (the same program has different
+representations for different architectures) or stable (it changes over time as optimization and
+language requirements change). It has representations for a huge variety of information that is
+useful for implementing mid-level compiler optimizations but is not useful for code generation (but
+which represents a large surface area for codegen implementers to deal with).  It also has undefined
+behavior (largely similar to that of C and C++) which makes some classes of optimization feasible or
+more powerful.  LLVM's binary format (bitcode) was designed for temporary on-disk serialization of
+the IR for link-time optimization, and not for stability or compressibility (although it does have
+some features for both of those).
+
+None of these problems are insurmountable. For example PNaCl defines a small portable
+[subset](https://developer.chrome.com/native-client/reference/pnacl-bitcode-abi) of the IR
+with reduced undefined behavior, and a stable version of the bitcode encoding. It also employs several
+techniques to improve startup performance. However, each customization, workaround, and special solution
+means less benefit from the common infrastructure. We believe that by taking our experience with LLVM and
+designing an IR and binary encoding for our goals and requirements, we can do much better than adapting a
+system desgined for other purposes.
+
+Note that this discussion applies to LLVM's IR and backend code. LLVM's clang frontend and midlevel
```

Comment:
Being pedantic, I suggest making it clear that we're just talking about LLVM IR in the context of being the standardized format. LLVM IR is used in clang and the midlevel optimizers and that's fine. And for the same reasons discussed earlier, I recommend omitting mention of the backend code here.